runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -110,65 +118,45 @@ NpcIndicatorsConfig provideConfig(ConfigManager configManager)
 	protected void startUp() throws Exception
 	{
 		keyManager.registerKeyListener(inputListener);
+		highlights = getHighlights();
+		rebuildNpcs();
 	}
 
 	@Override
 	protected void shutDown() throws Exception
 	{
 		npcTags.clear();
-		taggedNpcs.clear();
 		keyManager.unregisterKeyListener(inputListener);
 	}
 
 	@Subscribe
-	public void onFocusChanged(FocusChanged focusChanged)
-	{
-		// If you somehow manage to right click while holding shift, then click off screen
-		if (!focusChanged.isFocused() && hotKeyPressed)
-		{
-			updateNpcMenuOptions(false);
-		}
-	}
-
-	@Subscribe
-	public void onMenuObjectClicked(MenuOptionClicked click)
-	{
-		if (click.getMenuOption().equals(TAG))
-			toggleTag(click.getId());
-	}
-
-	@Subscribe
-	public void onGameTick(GameTick tick)
+	public void onConfigChanged(ConfigChanged configChanged)
 	{
-		highlightedNpcs = buildNpcsToHighlight();
-		taggedNpcs.clear();
-		if (npcTags.isEmpty() || !config.isTagEnabled())
+		if (!configChanged.getGroup().equals("npcindicators"))
 		{
 			return;
 		}
-		for (NPC npc : client.getNpcs())
-		{
-			if (npcTags.contains(npc.getIndex()) && npc.getName() != null)
-			{
-				taggedNpcs.add(npc);
-			}
-		}
-	}
 
-	@Override
-	public Collection<Overlay> getOverlays()
-	{
-		return Arrays.asList(npcClickboxOverlay, npcMinimapOverlay);
+		highlights = getHighlights();
+		rebuildNpcs();
 	}
 
-	private Map<NPC, String> buildNpcsToHighlight()
+	private List<String> getHighlights()
 	{
 		String configNpcs = config.getNpcToHighlight().toLowerCase();
 		if (configNpcs.isEmpty())
-			return Collections.EMPTY_MAP;
+			return Collections.EMPTY_LIST;
```

Comment:
Use Collections.emptyList() instead of EMPTY_LIST (avoids unchecked warning, all that emptyList method does is cast it inside)